src/lib/pipeline.ml

```diff
@@ -84,9 +87,14 @@ type _ t =
   | Bam_pair: bam t * bam t -> bam_pair t
   | Somatic_variant_caller: somatic Variant_caller.t * bam_pair t -> vcf t
   | Germline_variant_caller: germline Variant_caller.t * bam t -> vcf t
+  | Rna_hla_typer: fastq_sample t -> hla_types t
```

Comment:
`Somatic_variant_caller` is a "factorization" of all the variant-callers 9just to make the GADT a bit less crazy), the constructors in `Construct` still use the name of the variants callers and `somatic Variant_caller.t` contains the SVC-specific info (including the "name" that will appear in the JSON).

We could do the same factorization for aligners indeed.